Mage/src/main/java/mage/abilities/costs/mana/ManaCostsImpl.java

```diff
@@ -143,7 +143,9 @@ public boolean pay(Ability ability, Game game, UUID sourceId, UUID controllerId,
         }
 
         Player player = game.getPlayer(controllerId);
-        assignPayment(game, ability, player.getManaPool(), this);
+        if (!player.getManaPool().isForcedToPay()) {
+            assignPayment(game, ability, player.getManaPool(), this);
+        }
```

Comment:
@LevelX2 is calling assignPayment here (and the equivalent method in ManaCostImpl that seems to be unused) necessary for anything? (since it gets called immediately afterwards again) Seems like it potentially could be the source of those strange "floating mana gets auto-assigned to spell" pseudo-bugs I've encountered with stuff like Panglacial Wurm.